### PR TITLE
[#1044] chore: Add LICENSE-binary

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add Notice-binary

### Why are the changes needed?
Fix: #1044
If we want to release binary package, we should have LICENSE-binary. This is very important. The IPMC will check these licenses. 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No need.

### CODE REVIEW

File: LICENSE-binary [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+javax.servlet:javax.servlet-api
+
+Eclipse Public License (EPL) 2.0
+--------------------------------
+org.junit.platform:junit-platform-launcher
+org.junit.jupiter:junit-jupiter
+
+Common Public License (CPL) 1.0
+--------------------------------
+com.github.stefanbirkner:system-rule
```
Comment (LuciferYang) [inline LICENSE-binary:292]: need a empty line

File: LICENSE-binary [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+com.fasterxml.jackson.core:jackson-core
+com.fasterxml.jackson.core:jackson-databind
+com.fasterxml.jackson.module:jackson-module-scala_2.12
+info.picocli:picocli
+org.apache.directory.server:apacheds-kerberos-codec
+org.apache.hadoop:hadoop-common
+org.apache.hadoop:hadoop-mapreduce-client-app
+org.apache.hadoop:hadoop-mapreduce-client-core
+org.apache.hadoop:hadoop-mapreduce-examples
+org.apache.hadoop:hadoop-minicluster
```
Comment (LuciferYang) [inline LICENSE-binary:234]: Will `hadoop-minicluster`, `hadoop-minikdc` and `grpc-testing` also be included in the release package?

File: LICENSE-binary [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+BSD 3-Clause
+------------
+com.google.protobuf:protobuf-java
+com.google.protobuf:protobuf-java-util
+
+MIT License
+-----------
+org.slf4j:slf4j-api
+org.slf4j:slf4j-reload4j
+org.mockito:mockito-core
```
Comment (LuciferYang) [inline LICENSE-binary:275]: Likewise, if mockito is only used for testing, it should be possible to omit it from this file.

File: LICENSE-binary [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+Common Development and Distribution License (CDDL) 1.1
+------------------------------------------------------
+javax.annotation:javax.annotation-api
+javax.xml.bind:jaxb-api
+javax.servlet:javax.servlet-api
+
+Eclipse Public License (EPL) 2.0
+--------------------------------
+org.junit.platform:junit-platform-launcher
```
Comment (LuciferYang) [inline LICENSE-binary:287]: ditto

File: pom.xml [hunk 5]
```diff
@@ -232,6 +238,12 @@
         <groupId>com.google.guava</groupId>
         <artifactId>guava</artifactId>
         <version>${guava.version}</version>
+        <exclusions>
+          <exclusion>
```
Comment (jerqi) [inline pom.xml:242]: This jar's license contains multiple licenses. It seems to be used as test code. So I exclude it.

File: LICENSE-binary [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+org.eclipse.jetty:jetty-servlet
+org.eclipse.jetty:jetty-util
+org.eclipse.jetty:jetty-http
+org.eclipse.jetty:jetty-io
+org.eclipse.jetty:jetty-security
+commons-cli:commons-cli
+org.apache.httpcomponents:httpclient
+net.jpountz.lz4:lz4
+com.google.guava:guava
+com.google.guava::listenablefuture
```
Comment (LuciferYang) [inline LICENSE-binary:219]: `::`?

Comment (jerqi) [reply LICENSE-binary]: fixed.

**Comment by jerqi** [comment]
@LuciferYang Could we merge this pr first?

**Review by LuciferYang — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master. Thanks @jerqi

**Pull request merged**